### PR TITLE
Add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by jpkrohling**
Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+```
+make manifests docker-build docker-push deploy
+```
+
+By default, it will generate an image following the format `quay.io/${USER}/opentelemetry-operator:${VERSION}`. You can set the following env vars in front of the `make` command to override parts or the entirety of the image:
+
+* `USER`, to override the namespace
+* `VERSION`, to override only the version part
+* `IMG_PREFIX`, to override the registry, namespace and image name (`quay.io/${USER}/opentelemetry-operator`)
```
Comment (objectiser) [inline CONTRIBUTING.md:57]: Not necessarily a docs issue, but this doesn't seem to be a prefix, when it represents a majority of the image specification?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+Tests can also be run without an existing cluster. For that, install [`kubebuilder`](https://book.kubebuilder.io/quick-start.html#installation). In this case, the tests will bootstrap `etcd` and `kubernetes-api-server` for the tests. Run against an existing cluster whenever possible, though.
+
+## Project Structure
+
+Here's a general overview of the directories from this operator and what to expect in each one of them:
+
+```
+.
+├── api               # the CRDs that are handled by this operator
```
Comment (objectiser) [inline CONTRIBUTING.md:77]: Useful, but wondering if this could become outdated, as changes were necessarily get reflected. Possibly just the top level folders?

File: Makefile [hunk 3]
```diff
@@ -17,7 +17,7 @@ endif
 BUNDLE_METADATA_OPTS ?= $(BUNDLE_CHANNELS) $(BUNDLE_DEFAULT_CHANNEL)
 
 # Image URL to use all building/pushing image targets
-IMG_PREFIX ?= quay.io/${USER}
+IMG_PREFIX ?= quay.io/${USER}/opentelemetry-operator
```
Comment (objectiser) [inline Makefile:20]: Maybe this should be moved into `IMG`, so the prefix remains `quay.io/${USER}`?

File: Makefile [hunk 4]
```diff
@@ -1,5 +1,5 @@
 # Current Operator version
-VERSION ?= "$(shell git describe --tags | grep -Po "([\d\.]+)")" ## version, without the 'v' prefix
+VERSION ?= "$(shell git describe --tags | sed 's/^v//')"
```
Comment (objectiser) [inline Makefile:2]: Shouldn't this be changed in a separate PR?

File: CONTRIBUTING.md (hunk 2 above)
Comment (jpkrohling) [reply CONTRIBUTING.md:77]: Yeah, I was thinking the same. Perhaps godoc at the package level? I'll leave it here for now, but will create an issue to work on godoc.

File: Makefile (hunk 3 above)
Comment (jpkrohling) [reply Makefile:20]: IMG includes the version too, which we might not want to override while testing.

File: CONTRIBUTING.md (hunk 2 above)
Comment (jpkrohling) [reply CONTRIBUTING.md:77]: https://github.com/open-telemetry/opentelemetry-operator/issues/104

File: CONTRIBUTING.md (hunk 1 above)
Comment (jpkrohling) [reply CONTRIBUTING.md:57]: Something like this?

```
IMG_PREFIX = "quay.io"
IMG_NAMESPACE = "${USER}"
IMG_REPO = "opentelemetry-operator"
IMG = "${IMG_PREFIX}/${IMG_NAMESPACE}/${IMG_REPO}:${VERSION}"
```

File: Makefile (hunk 3 above)
Comment (objectiser) [reply Makefile:20]: I think I still prefer the scope of the previous `IMG_PREFIX` value :smile:

File: CONTRIBUTING.md (hunk 1 above)
Comment (objectiser) [reply CONTRIBUTING.md:57]: Yeah better +1

File: Makefile (hunk 4 above)
Comment (jpkrohling) [reply Makefile:2]: Created #105

File: CONTRIBUTING.md (hunk 1 above)
Comment (jpkrohling) [reply CONTRIBUTING.md]: Done, but left the IMG_NAMESPACE out, as it can be overridden already

**Comment by jpkrohling** [comment]
@objectiser is this OK to be merged?

**Review by objectiser — APPROVED** [review]
